Feature/thumbnails

mgdaily opened the pull request:
These are the ingester changes to support ingesting thumbnails to the science archive. The main change is to redirect the thumbnail payload to the /thumbnails endpoint if the file is a type defined in the ocs_archive as a thumbnail type. This relies on some small ocs_archive changes that I'll open a PR for.

The client side looks like this:

```python
    thumbnail_filename = '/Users/mdaily/Documents/ingester/thumbnail_testing/tfn0m419-sq32-20240426-0097-e91-small.jpg'
    fits_filename = '/Users/mdaily/Documents/ingester/thumbnail_testing/tfn0m419-sq32-20240426-0097-e91.fits.fz'
    # get header metadata from the main FITS image
    metadata = dict(fits.getheader(fits_filename, extname='SCI'))
    metadata['frame_basename'] = os.path.basename(fits_filename).split('.')[0]
    metadata['size'] = 'small'

    with open(thumbnail_filename, 'rb') as f:
        record = upload_file_and_ingest_to_archive(f, file_metadata=metadata)
```

Commit:
Add support for thumbnails. Add test jpg and update tests.

Also rename ingest_archive_record to ingest_archive_frame_record as it works with the /frames endpoint

Commit:
WIP on ingester changes for thumbnails

Commit:
Remove filename from record.

We already get the basename and the file extension, so we can always construct the filename if needed.

Commit:
Quick polish fixes to ingester lib

Commit:
Merge branch 'feature/thumbnails' of https://github.com/observatorycontrolsystem/ingester into feature/thumbnails

Commit:
Fix unused import, include frame_basename in metadata - we're not using frame_filename

jnation3406 commented on ocs_ingester/ingester.py:
```diff
@@ -158,7 +159,7 @@ def upload_file_to_file_store(fileobj, path=None, file_metadata=None):
 
 
-def ingest_archive_record(version, record, api_root=ingester_settings.API_ROOT,
+def ingest_archive_frame_record(version, record, api_root=ingester_settings.API_ROOT,
```
This rename probably also affects the shipper and ocs_example as well.

jnation3406 commented on ocs_ingester/ingester.py:
```diff
@@ -281,6 +282,7 @@ def ingest(self):
             raise NonFatalDoNotRetryError('Version with this md5 already exists')
 
         # Upload the file to s3 and get version information back
+        # we need to make sure that 
```
this comment seems incomplete

mgdaily replied:
D'oh. Sorry

jnation3406 commented on ocs_ingester/ingester.py:
```diff
@@ -293,4 +295,5 @@ def ingest(self):
         record['basename'] = self.datafile.open_file.basename
-        return self.archive.post_frame(record)
+
+        return self.archive.post_thumbnail(record) if self.datafile.open_file.extension in archive_settings.THUMBNAIL_FILETYPES else self.archive.post_frame(record)
```
I'm not sure about this... The archive is supposed to be flexible so imagine you are using the archive to store your raw image files (not fits)... that should work but this new logic can't differentiate between a thumbnail and "full image" that is also a jpeg. I think it might be better to make it explicit when ingesting that this is a small/large/whatever thumbnail when ingesting rather than infer it? I.e. the caller of the ingester should know, so they can decide.

mgdaily replied:
That makes a lot of sense. Let me re-think that. I was trying to minimize client changes, but inference is fragile.

jnation3406 reviewed (changes requested):
What do you think about having the caller of the ingester function specify if its a thumbnail and what "size", rather than inferring it?

Commit:
Rely on the client to specify if a data product is a thumbnail.

This is less fragile than inferring it, and now uses the ThumbnailDataFile class.

Commit:
Merge branch 'feature/thumbnails' of github.com:observatorycontrolsystem/ingester into feature/thumbnails

Commit:
Revert rename of ingest_archive_record.

Commit:
Update setup file, remove unnecessary test image

Commit:
Remove path to JPG

jnation3406 reviewed (approved):
Looks good now!

Commit:
Revert "Remove path to JPG"

This reverts commit 5fbf871c23d567c12c439d4e31d9d9e3b665065e.

Commit:
Revert "Update setup file, remove unnecessary test image"

This reverts commit 6e02d63069d3f43e15633879130872445c1ba8d6.

Commit:
Undo some commits - did not mean to delete the JPG and path information for tests.

Commit:
Make sure we set required metadata correctly if the user is uploading a thumbnail

Commit:
Submit coveralls with coveralls GH action to fix 422 errors

Commit:
Export coverage as xml for coveralls to properly process

Commit:
Fix run command

Commit:
Try format option to fix coveralls issues

Commit:
Fix codacy issues

Commit:
Update changelog
